src/ConstructiveSolidGeometry/SurfacePrimitives/SphereMantle.jl

```diff
@@ -0,0 +1,23 @@
+struct SphereMantle{T} <: AbstractSurfacePrimitive{T}
+    r::T
+end
+
+in(p::AbstractCoordinatePoint, s::SphereMantle) = _isapprox_sph_r(p, s.r)
+
+#=
+function sample(s::SphereMantle{T}, step::Real)::Vector{CylindricalPoint{T}} where {T}
+    samples = [
+        CylindricalPoint{T}(sqrt(s.r^2 - z^2),φ,z)
+        for z in in -s.r:step:s.r
+        for φ in 0:step/s.r:2π
+    ]
+end
+=#
+
+function sample(s::SphereMantle{T}, Nsamps::NTuple{3,Int})::Vector{CylindricalPoint{T}} where {T}
```

Comment:
We would also need `sample` methods that work with `CartesianTuple` and `CylindricalTuple`.

Reply:
I left them for you since you are familiar with what you need for the grid. Could you tackle these please? These sample methods are also missing from the Toroidal Surface Primitives

Reply:
Yes, I can take care of this.